swarm/storage/chunker.go

```diff
@@ -157,10 +142,15 @@ func (self *TreeChunker) Split(data io.Reader, size int64, chunkC chan *Chunk, s
 		close(errC)
 	}()
 
-	//TODO: add a timeout
-	if err := <-errC; err != nil {
-		close(quitC)
-		return nil, err
+
+	defer close(quitC)
+	select {
+	case err := <-errC:
+		if err != nil {
+			return nil, err
+		}
+	case <-time.NewTimer(splitTimeout).C:
+
```

Comment:
Right now it is a timeout exception to avoid routine leaks.... 
I plan to bring in Context for chunker in general while doing the "Concurrency" task.

Reply:
Ok at the very least it should log something then to help us identify such leaks (but I really think it should return an error).

Reply:
done. returning errOperationTimedOut